tests/unit/test_jlink.py

```diff
@@ -935,9 +940,10 @@ def test_jlink_close_context_manager(self):
           ``None``
         """
         # Use open_tunnel=None to avoid implicitly opening the connection.
-        with jlink.JLink(self.lib, open_tunnel=None):
+        with jlink.JLink(self.lib, open_tunnel=None) as context_jlink:
             self.dll.JLINKARM_OpenEx.assert_not_called()
             self.dll.JLINKARM_Close.assert_not_called()
+            context_jlink._jlink_open = True
```

Comment:
I'm not sure if this is quite right.  What is the set here doing?

Reply:
This has changed with latest commit.